test/Interop/Cxx/operators/Inputs/member-inline.h

```diff
@@ -147,4 +147,38 @@ struct NonReferenceReadIntArray {
   }
 };
 
+struct ReadOnlyIntArrayByVal {
+  ReadOnlyIntArrayByVal(int first) { values[0] = first; }
+  ReadOnlyIntArrayByVal(const ReadOnlyIntArray &other) {}
+  int operator[](int x) const { return values[x]; }
```

Comment:
Do we have any tests where the result of the subscript is non-trivial?

Reply:
Ah yeah, I will add those. Do you think it makes sense to add that for the reference case or no because it's just using a reference?

Reply:
I don't think it would hurt to have it for the reference case as well. At least there we know it's always going to be a pointer so there's a lower chance we get it wrong.